### PR TITLE
python311Packages.textual-dev: init at 1.2.1

### PR DESCRIPTION
**Pull request by YanniPapandreou**
## Description of changes

This adds the Python package `textual-dev` with homepage [here](https://github.com/Textualize/textual-dev). This Python package provides development tools for the Python package `textual` (homepage [here](https://github.com/Textualize/textual)) which is already packaged in nix (source [here](https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/development/python-modules/textual/default.nix#L80)).

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

## Notes:

This package also provides a cli binary `textual`. I am new to textual and only tested out a few commands with the resulting binary file in `./result/bin/`. I have not tested all of the functionality yet but not sure what is required; this is my first time contributing to nixpkgs.

### CODE REVIEW

File: pkgs/development/python-modules/textual-dev/default.nix [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    textual
+    typing-extensions
+  ];
+
+  pythonImportsCheck = [
+    "textual_dev"
+  ];
+
+  meta = with lib; {
+    description = " Development tools for Textual ";
```
Comment (MikaelFangel) [inline pkgs/development/python-modules/textual-dev/default.nix:44]:
```suggestion
    description = "Development tools for Textual";
```

File: pkgs/development/python-modules/textual-dev/default.nix [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    homepage = "https://github.com/Textualize/textual-dev";
+    license = licenses.mit;
```
Comment (MikaelFangel) [inline pkgs/development/python-modules/textual-dev/default.nix:46]: There should be a maintainers tag wlth a malntainer in it.

File: pkgs/development/python-modules/textual-dev/default.nix [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+buildPythonPackage rec {
+  pname = "textual-dev";
+  version = "1.2.1";
+  pyproject = true;
+
+  disabled = pythonOlder "3.7";
+
+  src = fetchFromGitHub {
+    owner = "Textualize";
+    repo = pname;
```
Comment (MikaelFangel) [inline pkgs/development/python-modules/textual-dev/default.nix:22]: The repo name and pname are not considered the same on nixpkgs (:

```suggestion
    repo = "textual-dev";
```

**Review by MikaelFangel — COMMENTED** [review]
Here are some general comments about the derivation 😊

**EDIT**
Also make sure you follow the contribution guidelines for commit messages (:

**Comment by YanniPapandreou** [comment]
Thanks for the comments and suggestions :)

> **EDIT** Also make sure you follow the contribution guidelines for commit messages (:

I am a bit unsure about how to change the existing commit message, I found this link [here](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/changing-a-commit-message#amending-older-or-multiple-commit-messages) but it seems discouraged.

File: pkgs/development/python-modules/textual-dev/default.nix (hunk 1 above)
Comment (YanniPapandreou) [reply pkgs/development/python-modules/textual-dev/default.nix:44]: Ahh missed this sorry, fixed

File: pkgs/development/python-modules/textual-dev/default.nix (hunk 2 above)
Comment (YanniPapandreou) [reply pkgs/development/python-modules/textual-dev/default.nix:46]: can I make myself the maintainer? I noticed maintainer tags on other packages but not sure if I can put myself or the syntax for this

File: pkgs/development/python-modules/textual-dev/default.nix (hunk 3 above)
Comment (YanniPapandreou) [reply pkgs/development/python-modules/textual-dev/default.nix:22]: I am not sure I follow this. I was basing my derivation on this one [here](https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/development/python-modules/textual/default.nix#L80) of textual. In that derivation they use `pname` for the `repo` name?

File: pkgs/development/python-modules/textual-dev/default.nix (hunk 2 above)
Comment (MikaelFangel) [reply pkgs/development/python-modules/textual-dev/default.nix:46]: You can add yourself as a maintainer, but be sure to read the contribution guidelines and add yourself to the maintainers list (maintainers.nix), which has a guide on how to do it at the beginning of the file.

File: pkgs/development/python-modules/textual-dev/default.nix (hunk 3 above)
Comment (MikaelFangel) [reply pkgs/development/python-modules/textual-dev/default.nix:22]: A lot of packages write it with pname, but it is actually not considered the same and hence my suggestion. (Based on multiple reviews that I've received)

**Comment by MikaelFangel** [comment]
To change your commit messages, you can use `git rebase -i HEAD~n` where n is the number of commits backwards you want to rebase.

File: pkgs/development/python-modules/textual-dev/default.nix (hunk 1 above)
Comment (YanniPapandreou) [reply pkgs/development/python-modules/textual-dev/default.nix]: fixed this in original commit when rebasing

**Comment by YanniPapandreou** [comment]
> To change your commit messages, you can use `git rebase -i HEAD~n` where n is the number of commits backwards you want to rebase. Then when you have rebased you need to force push the changes `git push --force-with-lease`.

thanks this works :)

File: pkgs/development/python-modules/textual-dev/default.nix (hunk 2 above)
Comment (YanniPapandreou) [reply pkgs/development/python-modules/textual-dev/default.nix:46]: thanks, I've done this :)

File: pkgs/development/python-modules/textual-dev/default.nix (hunk 3 above)
Comment (YanniPapandreou) [reply pkgs/development/python-modules/textual-dev/default.nix]: Ah okay thanks again, I've set `repo = "textual-dev"` explicitly

**Comment by YanniPapandreou** [comment]
Thanks for all the comments, I've incorporated these changes, please let me know if there is anything else :)

**Comment by MikaelFangel** [comment]
All changes you have made to init the package should be squashed, so when all looks correct you have 2 commits.
```
maintainers: add <handle>
<package>: init at x.x.x
```

and then the pr title should be `package: init at x.x.x`

**Comment by YanniPapandreou** [comment]
> All changes you have made to init the package should be squashed, so when all looks correct you have 2 commits.
> 
> ```
> maintainers: add <handle>
> <package>: init at x.x.x
> ```

Does the order of these two commits matter; otherwise done!

**Comment by MikaelFangel** [comment]
It does matter, as you will get an error from ofBorg with this order.

**Comment by YanniPapandreou** [comment]
I think I've managed to swap these thanks :)

File: pkgs/development/python-modules/textual-dev/default.nix [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  pname = "textual-dev";
+  version = "1.2.1";
+  pyproject = true;
+
+  disabled = pythonOlder "3.7";
+
+  src = fetchFromGitHub {
+    owner = "Textualize";
+    repo = "textual-dev";
+    rev = "fb5aa869784895066dbd0ec850d92ac9205cde73";
```
Comment (natsukium) [inline pkgs/development/python-modules/textual-dev/default.nix:23]: Looking at [the release date of pypi](https://pypi.org/project/textual-dev/), I think we should adopt this commit.
https://github.com/Textualize/textual-dev/commit/6afa9013a42cb18e9105e49d6a56874097f7c812

Then could you please add a brief comment about why we don't use a tag, and if possible, ask the upstream to add tags?

File: pkgs/development/python-modules/textual-dev/default.nix [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  ];
+
+  propagatedBuildInputs = [
+    aiohttp
+    click
+    msgpack
+    textual
+    typing-extensions
+  ];
+
```
Comment (natsukium) [inline pkgs/development/python-modules/textual-dev/default.nix:38]: Could you please enable tests with pytestCheckHook?
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#testing-python-packages-testing-python-packages

File: pkgs/development/python-modules/textual-dev/default.nix (hunk 4 above)
Comment (YanniPapandreou) [reply pkgs/development/python-modules/textual-dev/default.nix]: I've added a comment on my latest push and also asked upstream [here](https://github.com/Textualize/textual-dev/issues/26)

File: pkgs/development/python-modules/textual-dev/default.nix (hunk 5 above)
Comment (YanniPapandreou) [reply pkgs/development/python-modules/textual-dev/default.nix:42]: Done I believe, and all the tests passed :)

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 267542` at f6df515b run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>python310Packages.textual-dev</li>
    <li>python311Packages.textual-dev</li>
  </ul>
</details>

**Review by natsukium — APPROVED** [review]
LGTM, thanks for your contributions!
Welcome to nixpkgs!🎉

**Comment by YanniPapandreou** [comment]
Thanks! Also thanks for all the help improving this :)

**Pull request merged**